### ISSUE TITLE
BUG: test with declared minimal dependencies

### ISSUE THREAD

**Issue by bsipocz** [gammapy/gammapy#4001]
I don't see mpl 3.4 is being tested, even though it should be supported according to `setup.py`. One test run into issues with it though:

```
$  pytest --doctest-modules gammapy/visualization

Gammapy test data availability:
gammapy-data ... no
Gammapy environment variables:
GAMMAPY_DATA = not set
Setting matplotlib backend to "agg" for the tests.
================================================= test session starts ==================================================
platform darwin -- Python 3.9.1, pytest-7.1.2, pluggy-0.13.1
WARNING: AstropyDeprecationWarning: The ignore_warnings class is deprecated and may be removed in a future version.
        Use pytest.mark.filterwarnings instead. [pytest_astropy_header.display]
WARNING: AstropyDeprecationWarning: The catch_warnings class is deprecated and may be removed in a future version.
        Use pytest.warns instead. [astropy.tests.helper]

Running tests with Astropy version 5.1.
Running tests in gammapy/visualization.

Date: 2022-06-17T17:04:03

Platform: macOS-10.16-x86_64-i386-64bit

Executable: /Users/bsipocz/.pyenv/versions/3.9.1/bin/python3.9

Full Python Version: 
3.9.1 (default, Jan 11 2021, 15:50:22) 
[Clang 10.0.0 (clang-1000.11.45.5)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.23.0rc3
Scipy: 1.5.4
Matplotlib: 3.4.0
h5py: 3.1.0
Pandas: 1.1.5
cython: 0.29.24
iminuit: 2.11.2
matplotlib: 3.4.0
astropy: 5.1
regions: 0.6
healpy: 1.15.2
sherpa: not available
gammapy: 0.20.2.dev3+g5fdfee926.d20220617
naima: not available

Using Astropy options: remote_data: any.

rootdir: /Users/bsipocz/munka/devel/gammapy, configfile: setup.cfg
plugins: anyio-3.6.1, arraydiff-0.3, remotedata-0.3.2, requests-mock-1.9.3, mock-3.7.0, nbval-0.9.6, astropy-0.10.0, hypothesis-6.3.1, filter-subpackage-0.1.1, openfiles-0.5.0, doctestplus-0.11.1, astropy-header-0.1.2, cov-3.0.0
collected 8 items                                                                                                      

gammapy/visualization/cmap.py ..                                                                                 [ 25%]
gammapy/visualization/tests/test_cmap.py ..                                                                      [ 50%]
gammapy/visualization/tests/test_panel.py .                                                                      [ 62%]
gammapy/visualization/tests/test_utils.py F..                                                                    [100%]

======================================================= FAILURES =======================================================
_______________________________________ test_plot_spectrum_datasets_off_regions ________________________________________

    def test_plot_spectrum_datasets_off_regions():
        from gammapy.datasets import SpectrumDatasetOnOff
        from gammapy.maps import Map, RegionNDMap
    
        counts_off_1 = RegionNDMap.create("icrs;circle(0, 0.5, 0.2);circle(0.5, 0, 0.2)")
    
        counts_off_2 = RegionNDMap.create("icrs;circle(0.5, 0.5, 0.2);circle(0, 0, 0.2)")
    
        counts_off_3 = RegionNDMap.create("icrs;point(0.5, 0.5);point(0, 0)")
    
        m = Map.from_geom(geom=counts_off_1.geom.to_wcs_geom())
        ax = m.plot()
    
        dataset_1 = SpectrumDatasetOnOff(counts_off=counts_off_1)
    
        dataset_2 = SpectrumDatasetOnOff(counts_off=counts_off_2)
    
        dataset_3 = SpectrumDatasetOnOff(counts_off=counts_off_3)
    
        plot_spectrum_datasets_off_regions(
            ax=ax, datasets=[dataset_1, dataset_2, dataset_3]
        )
    
        print(ax.patches)
        print(len(ax.patches))
        assert_allclose(
>           ax.patches[0].get_edgecolor(), (0.121569, 0.466667, 0.705882, 1.0), rtol=1e-2
        )
E       IndexError: list index out of range

gammapy/visualization/tests/test_utils.py:40: IndexError
------------------------------------------------- Captured stdout call -------------------------------------------------
[]
0
=============================================== short test summary info ================================================
FAILED gammapy/visualization/tests/test_utils.py::test_plot_spectrum_datasets_off_regions - IndexError: list index ou...
============================================= 1 failed, 7 passed in 0.78s ==============================================
```

**Comment by adonath** [comment]
Double check after #4093.

**Comment by adonath** [comment]
The issue is still there: https://github.com/gammapy/gammapy/actions/runs/3146095272/jobs/5114152071#step:7:171
 I'll include a fix in #4116.
